Tests/UnitTests/ess-upload-file.component.spec.ts

```diff
@@ -74,69 +91,103 @@ const getNoEncData = () => {
-        component.showMessage(messageType,messageDesc);
-        expect(component.messageType).toEqual(messageType);
-        expect(component.messageDesc).toEqual(messageDesc);
+      component.displayErrorMessage = true;
+      component.showMessage(messageType, messageDesc);
+      expect(component.messageType).toEqual(messageType);
+      expect(component.messageDesc).toEqual(messageDesc);
     }
-    );
-
-    it('processEncFile should set encList',() => {
-      const file = new File([getEncData()], 'test.txt');
-      Object.defineProperty(file, 'type', { value: 'text/plain' });
-     component.encFile = file;
-     expect(component.validEncList).toBeUndefined();
-     component.processEncFile(getEncData());
-     expect(component.validEncList.length).toBe(1);
+  );
+
+  it.each`
+    fileType           |fileName           | encDataFunc          | expectedResult
+    ${'text/csv'}      |${'test.csv'}      | ${getEncData_csv()}  |  ${4}
+    ${'text/plain'}    |${'test.txt'}      | ${getEncData()}      |  ${1}
+    `('processEncFile should set encList',
+    ({ fileType, fileName, encDataFunc, expectedResult }: { fileType: 'text/csv' | 'text/permit'; fileName: string; encDataFunc: string; expectedResult: number }) => {
+      const file = new File([encDataFunc], fileName);
+      Object.defineProperty(file, 'type', { value: fileType });
+      component.encFile = file;
+      expect(component.validEncList).toBeUndefined();
+      component.processEncFile(encDataFunc);
+      expect(component.validEncList.length).toBe(expectedResult);
+    });
+  it.each`
+    fileType           |fileName           | encDataFunc          | expectedResult
+    ${'text/csv'}      |${'test.csv'}      | ${getEncData_csv()}  |  ${4}
+    ${'text/plain'}    |${'test.txt'}      | ${getEncData()}      |  ${1}
+    `('processEncFile should set encList',
+    ({ fileType, fileName, encDataFunc, expectedResult }: { fileType: 'text/csv' | 'text/permit'; fileName: string; encDataFunc: string; expectedResult: number }) => {
+      const file = new File([encDataFunc], fileName);
+      Object.defineProperty(file, 'type', { value: fileType });
+      component.encFile = file;
+      expect(component.validEncList).toBeUndefined();
+      component.processEncFile(encDataFunc);
+      expect(component.validEncList.length).toBe(expectedResult);
     });
 
-    it('processEncFile should set raise "Please upload valid ENC file." error',() => {
-      const file = new File([getEncData()], 'test.jpeg');
+  it.each`
+     encDataFunc
+	 ${getEncData_csv()}  
+     ${getEncData()}      
+    `('processEncFile should set raise "Please upload valid ENC file." error',
+    ({ encDataFunc }: { encDataFunc: string }) => {
+      const file = new File([encDataFunc], 'test.jpeg');
       Object.defineProperty(file, 'type', { value: 'image/jpeg' });
       component.encFile = file;
-      component.processEncFile(getEncData());
+      component.processEncFile(getEncData_csv());
```

Comment:
Please change to use `encDataFunc` instead of `getEncData_csv`

```suggestion
      component.processEncFile(encDataFunc());
```

Reply:
Updated. Thanks.